presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestRemoveIdentityProjectionsBelowProjection.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.planner.iterative.rule.test.BaseRuleTest;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.expression;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.assignment;
+import static com.facebook.presto.sql.planner.plan.AssignmentUtils.identityAssignments;
+
+public class TestRemoveIdentityProjectionsBelowProjection
+        extends BaseRuleTest
+{
+    @Test
+    public void test()
+    {
+        tester().assertThat(new RemoveIdentityProjectionsBelowProjection())
```

Comment:
This test case has non-limiting identity projections (input and outputs are exactly the same for the project).  My understanding is that was already covered by RemoveRedundantIdentityProjects and this case is more to cover cases where there is a project whose output is a subset of the input columns, and then some other project on top that is performing some function on the input e.g.
```
project ( a - > b + 1)
   project( b -> b)
         values (c, b)
```
becomes
```
project ( a -> b + 1)
    values(c, b)
```
Is my understanding correct?  If so, can you update the test to cover this case?

Reply:
Thanks! Yes, your understanding is correct.

Just updated to address the test.